Sources/Workspace/Workspace.swift

```diff
@@ -1768,28 +1809,46 @@ extension Workspace {
         }
     }
 
-    private func parseArtifacts(from manifests: DependencyManifests) throws -> (local: [ManagedArtifact], remote: [RemoteArtifact]) {
+    private func parseArtifacts(
+        from manifests: DependencyManifests
+    ) throws -> (
+        local: [ManagedArtifact],
+        localArchived: [LocalArchivedArtifact],
+        remote: [RemoteArtifact]
+    ) {
         let packageAndManifests: [(reference: PackageReference, manifest: Manifest)] =
             manifests.root.packages.values + // Root package and manifests.
             manifests.dependencies.map({ manifest, managed, _ in (managed.packageRef, manifest) }) // Dependency package and manifests.
 
         var localArtifacts: [ManagedArtifact] = []
+        var localArchivedArtifacts: [LocalArchivedArtifact] = []
         var remoteArtifacts: [RemoteArtifact] = []
 
         for (packageReference, manifest) in packageAndManifests {
             for target in manifest.targets where target.type == .binary {
                 if let path = target.path {
                     // TODO: find a better way to get the base path (not via the manifest)
                     let absolutePath = try manifest.path.parentDirectory.appending(RelativePath(validating: path))
-                    localArtifacts.append(
-                        .local(
-                            packageRef: packageReference,
-                            targetName: target.name,
-                            path: absolutePath)
-                    )
+                    
+                    if absolutePath.extension == LocalArchivedArtifact.extension {
+                        localArchivedArtifacts.append(
+                            LocalArchivedArtifact(
+                                packageRef: packageReference,
+                                targetName: target.name,
+                                archivePath: absolutePath,
+                                destinationPath: self.location.artifactsDirectory.appending(components: packageReference.name))
```

Comment:
in the local case we set the path to absolutePath which is computed off the relative path provided by the user in the manifest, in effort to reduce the API changes, maybe we can compute the destinationPath from that path also, i.e. 
`self.location.artifactsDirectory.appending(RelativePath(validating: path)` or something along these lines which can be computed and should be unique.

Reply:
Done, now `updateBinaryArtifacts` is the only function that underwent changes in `Wokrspace`.